File: app/src/main/java/com/example/rentalwheels/PaymentActivity.java
package com.example.rentalwheels;

import androidx.appcompat.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;


import butterknife.BindView;
import butterknife.ButterKnife;

public class PaymentActivity extends AppCompatActivity implements View.OnClickListener {

    private ProgressDialog mProgressDialog;

    @BindView(R.id.etAmount)
    EditText mAmount;
    @BindView(R.id.etPhone)EditText mPhone;
    @BindView(R.id.btnPay)
    Button mPay;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_payment);
        ButterKnife.bind(this);


        mProgressDialog = new ProgressDialog(this);

        mPay.setOnClickListener(this);


    }



    // Initiate payment after a user has entered their name and the amount they wish to pay.
    // The amount can also be passed in from the price of a product for example, if you were integrating an online shop.
    @Override
    public void onClick(View view) {
        if (view== mPay){
            String phone_number = mPhone.getText().toString();
            String amount = mAmount.getText().toString();
        }
    }
}

File: app/src/main/java/com/msdc/rentalwheels/modules/PaymentActivity.kt
package com.msdc.rentalwheels.modules

import android.app.ProgressDialog
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.msdc.rentalwheels.R

class PaymentActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var mProgressDialog: ProgressDialog


    lateinit var mAmount: EditText

    lateinit var mPhone: EditText

    lateinit var mPay: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_payment)

        mAmount = findViewById(R.id.etAmount)
        mPhone = findViewById(R.id.etPhone)
        mPay = findViewById(R.id.btnPay)


        mProgressDialog = ProgressDialog(this)

        mPay.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        if (view == mPay) {
            val phoneNumber = mPhone.text.toString()
            val amount = mAmount.text.toString()
            // Add your payment initiation logic here
        }
    }
}